app/src/main/java/com/beemdevelopment/aegis/vault/Vault.java

```diff
@@ -49,17 +70,51 @@ public static Vault fromJson(JSONObject obj) throws VaultException {
                 VaultEntry entry = VaultEntry.fromJson(array.getJSONObject(i));
                 entries.add(entry);
             }
+
+            if (obj.has("groups")) {
```

Comment:
Similarly to the importer, I think we should parse the groups first, then parse the entries and then check the groups they're in. With regards to the entries: I think we may have to move the check for the old ``group`` field here. The reason being: This is the only place where we can actually do the conversion to the new group system. Now we first add the entry and only then we check whether we need to convert it.

_(Having to do all this referential integrity checking manually really makes you wish we were using SQLite)_